app/models/ability.rb

```diff
@@ -76,8 +76,12 @@ def signed_in(user)
     can :manage, Event do |event|
       event.users.include?(user)
     end
-    # can create an event until the last day of a conference
-    can :create, Event, program_id: Conference.where('end_date >= ?', Date.today).map { |conference| conference.program.id}.compact
+
+    # cannot create an event if program does not have open cfp
+    cannot [:new, :create], Event do |event|
+      user_inclusion = event.event_users.map { |event_user| event_user.user.id }.compact.include? user.id
+      !event.program.cfp_open? || !event.new_record? || !user_inclusion
+    end
```

Comment:
I have to use combination of `can` followed by `cannot` (overrides previous can rule), cause:
```Ruby
can [:new, :create], Event do |event|
   event.program.cfp_open?
end
```
^ will logically OR the two rules, we want to authorize if `AND` rule passes.